sys-utils/nsenter.c

```diff
@@ -119,6 +122,38 @@
 static int env_fd = -1;
 static int uid_gid_fd = -1;
 
+static int set_parent_user_ns_fd(void)
+{
+	struct namespace_file *nsfile;
+	struct namespace_file *user_nsfile;
+	int parent_ns = -1;
+
+	for (nsfile = namespace_files; nsfile->nstype; nsfile++) {
+		if (nsfile->nstype == CLONE_NEWUSER) {
+			user_nsfile = nsfile;
+		}
+
+		if (nsfile->fd == -1)
+			continue;
+
+		parent_ns = ioctl(nsfile->fd, NS_GET_USERNS);
+		if (parent_ns < 0) {
+			err(EXIT_FAILURE, _("failed to open parent ns"));
+			return -1;
+		}
+
+		break;
+	}
+
+	if (parent_ns < 0) {
+		err(EXIT_FAILURE, _("no namespaces to get parents of"));
+		return -1;
+	}
+
+	user_nsfile->fd = parent_ns;
```

Comment:
This is a valid warning

Reply:
It is impossible to be not initialized because `namespace_files` is initialized with a literal and `nsfile->nstype == CLONE_NEWUSER` should always be present.